### DISCUSSION TITLE
Question about generic nullable, CA1508, CS8602

### DISCUSSION THREAD

**Discussion by Cyl18** [dotnet/csharplang#4399] [Q&A] [answered]
```csharp
    public sealed class ImmutableNullableContainer<T>
    {
        [MaybeNull]
        public T? Value { [return: MaybeNull] get; }

        public ImmutableNullableContainer(T? value)
        {
            Value = value;
        }

        public override string? ToString()
            => Value.ToString();
    }

```

![image](https://user-images.githubusercontent.com/14993992/107114815-c476a580-68a3-11eb-9400-38ecd856797c.png)
it shows CS8602.
if I add a `?` after `Value`, it shows CA1508:
![image](https://user-images.githubusercontent.com/14993992/107114845-f5ef7100-68a3-11eb-92e7-ab80bfdd9461.png)

And I can't pass `null` to the constructor:
![image](https://user-images.githubusercontent.com/14993992/107114921-541c5400-68a4-11eb-9c0e-9500c442a775.png)

* Version: .NET 5

**Comment by yaakov-h** [comment]
I'm not seeing the same errors.

https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwMRQgG0wQwEaYCmABIVPkQLABQNAAgEwCM9ADMXUwHQAiAljgDmUAPYBnYHwDGYrgGERcQgEFymAJ5i+YgNw16TAJwAKKIQDuxAJIBbGxGAVCAOSy4ChBVEd8zYADycrAB8pm4AlOF6tNR0AMzEYoQ4RHAcDNZ2Dk6u2E5ePn7+ACrBNADeNMTVxADaALI46ngubgC6VTXxxMUA/MQAaikQJOV1dADsIMSNza3YbcSChMA6xAC++tQ1HAm29o4eue5EBTi+hGDGfcQAbsOE4Z3Vlds7NUOYI8QAvHcP0R2mxiO26IlulzAfCUHCYrH6xREAGVgFCoIJjE83u9fsFBg9elxESi0RiojRgUA

What version of Visual Studio are you using?

**Reply by Cyl18** [reply]
16.8.4
and I have enabled all analyzers  `<AnalysisMode>AllEnabledByDefault</AnalysisMode>`

**Comment by 333fred** [comment, marked as answer]
> And I can't pass null to the constructor:

You can if T is actually resolved to a real type. You can also pass default at the location you showed.

As to the CA warning, that looks like a bug. Please file an issue on dotnet/roslyn-analyzers.

**Reply by Cyl18** [reply]
thanks!
